File: app/src/main/java/mx/yellowme/app/simple/UmbrellaAdapter.java
package mx.yellowme.app.simple;

import android.support.annotation.NonNull;

import mx.yellowme.waterproof.mvp.simple.SimpleAdapter;
import mx.yellowme.waterproof.mvp.simple.SimpleWaterproofFragment;

/**
 * Created by migdonio on 8/1/17.
 */

public class UmbrellaAdapter extends SimpleAdapter<Umbrella, UmbrellaViewHolder> {

    protected UmbrellaAdapter(
            @NonNull SimpleWaterproofFragment wpFragment,
            @NonNull UmbrellaViewHolder wpViewHolder
    ) {
        super(wpFragment, wpViewHolder);
    }

    @Override
    protected void bindViews(UmbrellaViewHolder mViewHolder) {
        mViewHolder.umbrellaPriceTextView.setText(String.valueOf(mItem.getPrice()));
        mViewHolder.umbrellaColorTextView.setText(mItem.getColor());
    }
}

File: app/src/main/java/mx/yellowme/app/simple/UmbrellaAdapter.kt
package mx.yellowme.app.simple

import mx.yellowme.waterproof.mvp.simple.SimpleAdapter
import mx.yellowme.waterproof.mvp.simple.SimpleWaterproofFragment

/**
 * Created by migdonio on 8/1/17.
 */

class UmbrellaAdapter(
        wpFragment: SimpleWaterproofFragment<*, *>,
        wpViewHolder: UmbrellaViewHolder
) : SimpleAdapter<Umbrella, UmbrellaViewHolder>(wpFragment, wpViewHolder) {

    override fun bindViews(mViewHolder: UmbrellaViewHolder) {
        mViewHolder.umbrellaPriceTextView.text = item!!.price.toString()
        mViewHolder.umbrellaColorTextView.text = item!!.color
    }
}
